lib/config-array-factory.js

```diff
@@ -495,7 +495,7 @@ class ConfigArrayFactory {
                 basePath
             );
 
-            if (fs.existsSync(ctx.filePath)) {
+            if (fs.existsSync(ctx.filePath) && fs.statSync(ctx.filePath).isFile()) {
```

Comment:
a similar check is done here:

https://github.com/eslint/eslint/blob/e837d811e9275e43ccc266aba10e9c0833e0376c/lib/init/npm-utils.js#L33